File: app/src/main/java/com/zhiyicx/thinksnsplus/modules/wallet/integration/detail/IntegrationDetailPresenter.java
package com.zhiyicx.thinksnsplus.modules.wallet.integration.detail;

import com.zhiyicx.baseproject.base.TSListFragment;
import com.zhiyicx.thinksnsplus.base.AppBasePresenter;
import com.zhiyicx.thinksnsplus.base.BaseSubscribeForV2;
import com.zhiyicx.thinksnsplus.data.beans.RechargeSuccessV2Bean;
import com.zhiyicx.thinksnsplus.data.source.repository.BillRepository;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import rx.Subscription;

/**
 * @Author Jliuer
 * @Date 2017/06/05/10:06
 * @Email Jliuer@aliyun.com
 * @Description
 */
public class IntegrationDetailPresenter extends AppBasePresenter<IntegrationDetailContract.View> implements IntegrationDetailContract.Presenter {


    @Inject
    BillRepository mBillRepository;

    @Inject
    public IntegrationDetailPresenter(IntegrationDetailContract.View rootView) {
        super(rootView);
    }

    @Override
    public void requestNetData(Long maxId, final boolean isLoadMore) {
        Subscription subscribe = mBillRepository.integrationOrdersSuccess(TSListFragment.DEFAULT_PAGE_SIZE,maxId.intValue(), mRootView.getChooseType(),
                mRootView
                .getBillType())
                .subscribe(new BaseSubscribeForV2<List<RechargeSuccessV2Bean>>() {
                    @Override
                    protected void onSuccess(List<RechargeSuccessV2Bean> data) {
                        mRootView.onNetResponseSuccess(data, isLoadMore);
                    }

                    @Override
                    protected void onFailure(String message, int code) {
                        super.onFailure(message, code);
                        mRootView.showMessage(message);
                    }

                    @Override
                    protected void onException(Throwable throwable) {
                        super.onException(throwable);
                        mRootView.onResponseError(throwable, isLoadMore);
                    }
                });
        addSubscrebe(subscribe);
    }

    @Override
    public void requestCacheData(Long maxId, boolean isLoadMore) {
        mRootView.onCacheResponseSuccess(new ArrayList<>(), isLoadMore);
    }

    @Override
    public boolean insertOrUpdateData(@NotNull List<RechargeSuccessV2Bean> data, boolean isLoadMore) {
        return true;
    }

    @Override
    public void selectBillByAction(int action) {
//        List<RechargeSuccessBean> data = mRechargeSuccessBeanGreenDao.selectBillByAction(action);
//        mRootView.onNetResponseSuccess(data, false);
    }

    @Override
    public void selectAll() {
        requestCacheData(1L, false);
    }


}

File: app/src/main/java/com/zhiyicx/thinksnsplus/modules/wallet/integration/detail/IntegrationDetailPresenter.kt
package com.zhiyicx.thinksnsplus.modules.wallet.integration.detail

import com.zhiyicx.baseproject.base.TSListFragment
import com.zhiyicx.thinksnsplus.base.AppBasePresenter
import com.zhiyicx.thinksnsplus.base.BaseSubscribeForV2
import com.zhiyicx.thinksnsplus.data.beans.RechargeSuccessV2Bean
import com.zhiyicx.thinksnsplus.data.source.repository.BillRepository
import java.util.*
import javax.inject.Inject

/**
 * @Describe
 * @Author Jungle68
 * @Date 2018/1/31
 * @Contact master.jungle68@gmail.com
 */
class IntegrationDetailPresenter @Inject
constructor(rootView: IntegrationDetailContract.View) : AppBasePresenter<IntegrationDetailContract.View>(rootView), IntegrationDetailContract.Presenter {


    @Inject
    lateinit var mBillRepository: BillRepository

    override fun requestNetData(maxId: Long?, isLoadMore: Boolean) {
        val subscribe = mBillRepository.integrationOrdersSuccess(TSListFragment.DEFAULT_PAGE_SIZE, maxId!!.toInt(), mRootView.mChooseType,
                mRootView
                        .billType)
                .subscribe(object : BaseSubscribeForV2<List<RechargeSuccessV2Bean>>() {
                    override fun onSuccess(data: List<RechargeSuccessV2Bean>) {
                        mRootView.onNetResponseSuccess(data, isLoadMore)
                    }

                    override fun onFailure(message: String, code: Int) {
                        super.onFailure(message, code)
                        mRootView.showMessage(message)
                    }

                    override fun onException(throwable: Throwable) {
                        super.onException(throwable)
                        mRootView.onResponseError(throwable, isLoadMore)
                    }
                })
        addSubscrebe(subscribe)
    }

    override fun requestCacheData(maxId: Long?, isLoadMore: Boolean) {
        mRootView.onCacheResponseSuccess(ArrayList(), isLoadMore)
    }

    override fun insertOrUpdateData(data: List<RechargeSuccessV2Bean>, isLoadMore: Boolean): Boolean {
        return true
    }

}
